src/cargo/util/config/mod.rs

```diff
@@ -1717,8 +1717,19 @@ impl Config {
     }
 
     pub fn http_config(&self) -> CargoResult<&CargoHttpConfig> {
-        self.http_config
-            .try_borrow_with(|| self.get::<CargoHttpConfig>("http"))
+        self.http_config.try_borrow_with(|| {
+            #[allow(unused_mut)]
+            let mut http = self.get::<CargoHttpConfig>("http")?;
+            #[cfg(target_os = "macos")]
+            {
+                // We only take care of macOS now, since it is one of the major
+                // platforms. For other OS we encourage not to stick to older
+                // versions of libcurl.
```

Comment:
I'm a little uncertain about making this macos only. What would be the concern with just checking this on all platforms?

Reply:
That was just me trying to shave off all old system libcurl aggressively 😅.

The constraint has been removed.

And weird. I forced pushed something but it didn't show up in this pull request.